edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/CloudConnectionProvider.cs

```diff
@@ -83,14 +86,20 @@ public async Task<Try<ICloudConnection>> Connect(IClientCredentials clientCreden
                 var cloudListener = new CloudListener(this.edgeHub.Expect(() => new InvalidOperationException("EdgeHub reference should not be null")), clientCredentials.Identity.Id);
                 string productInfo = await this.productInfoStore.GetEdgeProductInfo(clientCredentials.Identity.Id);
 
+                string authChain = string.Empty;
+                if (this.nestedEdgeEnabled)
+                {
+                    Option<string> authChainMaybe = await this.deviceScopeIdentitiesCache.GetAuthChain(clientCredentials.Identity.Id);
+                    authChain = authChainMaybe.Expect(() => new InvalidOperationException($"No auth chain for the client identity: {clientCredentials.Identity.Id}"));
+                }
+
                 // Get the transport settings
-                Option<string> authChain = await this.deviceScopeIdentitiesCache.GetAuthChain(clientCredentials.Identity.Id);
                 ITransportSettings[] transportSettings = GetTransportSettings(
                     this.upstreamProtocol,
                     this.connectionPoolSize,
                     this.proxy,
                     this.useServerHeartbeat,
-                    authChain.Expect(() => new InvalidOperationException($"No auth chain for the client identity: {clientCredentials.Identity.Id}")));
+                    authChain);
```

Comment:
So this will set the authChain to empty string? It seems like it will be cleaner to pass the option and set the authChain on the transport settings object only if it is not empty.

Reply:
for the next phase (updating identities connection-failure), we'll need to check the authchain here anyways.  So pushing that logic lower into the static helpers will actually make it more awkward.

The empty string path is strictly temporary until our service side and SDK dependencies come in, at which point this flag wouldn't be needed and we won't have pass in an empty string anymore.